### PR TITLE
Release v0.12.0

### LINKED ISSUE #2929: Scheduled eBPF release is due
This is an automated reminder that a new release for eBPF is scheduled to be created:

1. Follow instructions in [ReleaseProcess.md](https://github.com/microsoft/ebpf-for-windows/blob/main/docs/ReleaseProcess.md).
1. Sync the internal `mscodehub` and follow the internal release instructions in `\.internal\docs\ReleaseEbpfRedistProcess.md`.

### PR DESCRIPTION
**Pull request by gtrevi**
Closes #2929 

## Description

Update sources to v0.12.0.

## Testing

CI/CD

## Documentation

NIT updates to `ReleaseProcess.md`.

## Installation

n.a.

### CODE REVIEW

File: docs/ReleaseProcess.md [hunk 1]
```diff
@@ -14,13 +14,33 @@ eBPF for Windows, and how to service it later.
     # Replace "X.Y.Z" with the new version number being released
     .\scripts\update-release-version.ps1 X Y Z
     ```
+
+    a successful run of the script will produce similar output to the following:
```
Comment (dthaler) [inline docs/ReleaseProcess.md:18]:
```suggestion
    A successful run of the script will produce output similar to the following:
```

File: docs/ReleaseProcess.md [hunk 2]
```diff
@@ -14,13 +14,33 @@ eBPF for Windows, and how to service it later.
...
+    Generating output for atomic_instruction_fetch_add.o
+    ...
+    ...
+    Generating output for printk_unsafe.o
+    Expected 'bpf2c' output regenerated.
+    Please verify all the changes then submit the pull-request into the 'release/0.12' branch.
+    ```
 1. Verify all the changes then commit all in the working branch.
-    >NOTE: the formatting rules may complain about the formatting of the generated `.c` files from the script above, in this case, format them with the following:
+    >NOTE: the pre-commit formatting rules may complain about the formatting of the `.c` files generated above by the 'bpf2c' tool, in this case, format them with the following command:
```
Comment (dthaler) [inline docs/ReleaseProcess.md:36]:
```suggestion
    >Note: The pre-commit formatting rules may complain about the formatting of the `.c` files generated above by the 'bpf2c' tool. In this case, format them with the following command:
```

Comment (gtrevi) [reply docs/ReleaseProcess.md]: updated

File: docs/ReleaseProcess.md (hunk 1 above)
Comment (gtrevi) [reply docs/ReleaseProcess.md]: updated

**Review by saxena-anurag — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Review by saxena-anurag — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Comment by gtrevi** [comment]
Moving back to draft, as there's WIP investigation.

**Review by saxena-anurag — APPROVED** [review]

**Review by dthaler — APPROVED** [review]

**Pull request merged**